File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/CurrencyConversionMapper.cs
using CurrencyConversionSdk = OnlinePayments.Sdk.Domain.CurrencyConversion;
using CurrencyConversionDto = Business.Domain.Payments.PaymentDetails.CurrencyConversion;

namespace Infrastructure.Mappers.PaymentDetails;

public static class CurrencyConversionMapper
{
    public static CurrencyConversionDto Map(CurrencyConversionSdk? response)
    {
        return new()
        {
            AcceptedByUser = response?.AcceptedByUser,
            Proposal = DccProposalMapper.Map(response?.Proposal)
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/CurrencyConversionMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\CurrencyConversion as CurrencyConversionDto;
use OnlinePayments\Sdk\Domain\CurrencyConversion as CurrencyConversionSdk;

final class CurrencyConversionMapper
{
    public static function mapFromSdkResponse(?CurrencyConversionSdk $response): ?CurrencyConversionDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new CurrencyConversionDto();
        $dto->acceptedByUser = $response?->acceptedByUser ?? null;
        $dto->proposal = DccProposalMapper::mapFromSdkResponse($response?->proposal ?? null);

        return $dto;
    }
}
